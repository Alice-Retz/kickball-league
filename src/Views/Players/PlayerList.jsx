import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getPlayers } from '../../Services/players';

export default function PlayerList() {
	const [players, setPlayers] = useState([]);

	useEffect(() => {
		getPlayers().then((res) => setPlayers(res));
	}, []);

	return (
		<>
			<h1>Players:</h1>
			<ul>
				{players.map((player) => {
					return (
						<li key={player.id}>
							<Link to={`/players/${player.id}`}>{player.name}</Link>
						</li>
					);
				})}
			</ul>
		</>
	);
}
