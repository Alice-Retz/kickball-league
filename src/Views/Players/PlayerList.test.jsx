import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import PlayerList from './PlayerList';

it('displays a list of players', async () => {
	const { container } = render(
		<MemoryRouter initialEntries={['/players']}>
			<Route path='/players'>
				<PlayerList />
			</Route>
		</MemoryRouter>
	);

	const playerName = await screen.findByText('Mike Comeaux', { exact: false });

	expect(playerName).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
