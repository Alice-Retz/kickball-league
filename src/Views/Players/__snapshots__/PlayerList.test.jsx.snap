// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of players 1`] = `
<div>
  <h1>
    Players:
  </h1>
  <ul>
    <li>
      <a
        href="/players/4"
      >
        Babette Hebert
      </a>
    </li>
    <li>
      <a
        href="/players/1"
      >
        Boudreaux Doucet
      </a>
    </li>
    <li>
      <a
        href="/players/2"
      >
        Larry Broussard
      </a>
    </li>
    <li>
      <a
        href="/players/5"
      >
        Maggie Arcenaux
      </a>
    </li>
    <li>
      <a
        href="/players/3"
      >
        Mike Comeaux
      </a>
    </li>
    <li>
      <a
        href="/players/6"
      >
        Nicole Melan√ßon
      </a>
    </li>
  </ul>
</div>
`;
